File: algorithms/strings/is_alphabetical/is_alphabetical.py
import re

def is_alphabetical(stringArg):
    stringArg = stringArg.lower()
    strLength = len(stringArg)
    pattern = re.compile("[^a-z]")

    if not stringArg or strLength <= 1:
        return True
    if pattern.match(stringArg[0]):
        return "Invalid input"

    for i in range(0,strLength - 1):
        if pattern.match(stringArg[i+1]):
            return "Invalid input"
        if stringArg[i] > stringArg[i + 1]:
            return False
    return True


print(is_alphabetical("")); # true
print(is_alphabetical(" ")); # true
print(is_alphabetical("a")); # true
print(is_alphabetical("ab")); # true
print(is_alphabetical("AbC")); # true
print(is_alphabetical("abc")); # true
print(is_alphabetical("aaabbbc")); # true
print(is_alphabetical("abcedfg")); # false
print(is_alphabetical("facetiously")); # false
print(is_alphabetical("adept")); # true
print(is_alphabetical("about")); # false
print(is_alphabetical("abcde!")); # invalid input

File: algorithms/strings/is_alphabetical/is_alphabetical.js
function is_alphabetical(stringArg) {
  stringArg = stringArg.toLowerCase();
  var strLength = stringArg.length;

  if (!stringArg || strLength <= 1) {
    return true;
  }

  if (stringArg[0].match(/[^a-z]/)) {
    return "Invalid input";
  }

  for (var i = 0; i < strLength - 1; i++) {
    if (stringArg[i+1].match(/[^a-z]/)) {
      return "Invalid input";
    }
    if (stringArg[i] > stringArg[i + 1]) {
      return false;
    }
  }

  return true;
}

console.log(is_alphabetical("")); // true
console.log(is_alphabetical(" ")); // true
console.log(is_alphabetical("a")); // true
console.log(is_alphabetical("ab")); // true
console.log(is_alphabetical("AbC")); // true
console.log(is_alphabetical("abc")); // true
console.log(is_alphabetical("aaabbbc")); // true
console.log(is_alphabetical("abcedfg")); // false
console.log(is_alphabetical("facetiously")); // false
console.log(is_alphabetical("adept")); // true
console.log(is_alphabetical("about")); // false
console.log(is_alphabetical("abcde!")); // invalid input
